src/Microsoft.Azure.WebJobs.ServiceBus/Common/GenericBinder.cs

```diff
@@ -90,7 +90,59 @@ internal class GenericBinder
             {
                 // Default, assume a Poco
                 return new PocoTriggerArgumentBinding<TMessage, TTriggerValue>(hooks, converterManager, elementType);
-            }         
+            }
+        }
+
+        /// <summary>
+        /// Creates a binding that binds to an <see cref="IFlushCollector{T}"/> with the specified argument type. Allows for a binding
+        /// to be generated for any POCO type.
+        /// </summary>
+        /// <param name="parameter">The ParameterInfo for the binding</param>
+        /// <param name="collectorGenericType">The generic type that must implement <see cref="IFlushCollector{T}"/> and have a public constructor with at most one parameter.</param>
+        /// <param name="collectorGenericArgumentType">The generic argument type for the <see cref="IFlushCollector{T}"/></param>
+        /// <param name="converterManager">A converter manager to pass along.</param>        
+        /// <param name="invokeStringBinder">A <see cref="Func{T, TContext}"/> that returns the TContext to be used by the <see cref="IFlushCollector{T}"/></param>
+        /// <returns></returns>
+        public static IBinding BindGenericCollector<TContext>(ParameterInfo parameter, Type collectorGenericType, Type collectorGenericArgumentType,
+            IConverterManager converterManager, Func<string, TContext> invokeStringBinder)
+        {
+            if (!collectorGenericType.IsGenericTypeDefinition)
+            {
+                throw new ArgumentException("The parameter 'collectorGenericType' must be a generic type definition.");
+            }
+            if (collectorGenericType.GetInterface(typeof(IFlushCollector<>).Name) == null)
+            {
+                throw new ArgumentException("The Type specified by parameter 'collectorGenericType' must implement IFlushCollector<T>.");
+            }
+
+            Type flushCollectorInterfaceType = typeof(IFlushCollector<>).MakeGenericType(collectorGenericArgumentType);
+            Type actualCollectorType = collectorGenericType.MakeGenericType(collectorGenericArgumentType);
+
+            // Create a delegate to pass as the builder func to BindCollector
+            Type funcType = typeof(Func<,,>).MakeGenericType(typeof(object), typeof(ValueBindingContext), flushCollectorInterfaceType);
+            MethodInfo getCollector = typeof(GenericBinder).GetMethod("GetCollector", BindingFlags.NonPublic | BindingFlags.Static)
+                .MakeGenericMethod(actualCollectorType, collectorGenericArgumentType);
+            var del = Delegate.CreateDelegate(funcType, getCollector);
+
+            // Create the method and parameters.
+            MethodInfo bindCollectorMethod = typeof(GenericBinder).GetMethod("BindCollector").MakeGenericMethod(collectorGenericArgumentType, typeof(TContext));
+            object[] parameters = new object[] { parameter, converterManager, del, null, invokeStringBinder };
```

Comment:
I wouldn't block the change on this, but this parameter (invokeString) should be non-null.  We'll compute the initial TContext by calling invokeStringBinder(invokeString). See line 262.  
The idea is that the dashboard could replay this function and change the invokeString in the replayed instance.